src/index.js

```diff
@@ -122,83 +152,119 @@ class ExtractTextPlugin {
 
   apply(compiler) {
     const options = this.options;
+
     compiler.plugin('this-compilation', (compilation) => {
       const extractCompilation = new ExtractTextPluginCompilation();
-      compilation.plugin('normal-module-loader', (loaderContext, module) => {
-        loaderContext[NS] = (content, opt) => {
-          if (options.disable) {
-            return false;
-          }
-          if (!Array.isArray(content) && content != null) {
-            throw new Error(
-              `Exported value was not extracted as an array: ${JSON.stringify(
-                content
-              )}`
-            );
-          }
-          module[NS] = {
-            content,
-            options: opt || {},
+
+      compilation.hooks.normalModuleLoader.tap(
+        'normal-module-loader',
+        (loaderContext, module) => {
+          loaderContext[NS] = (content, opt) => {
+            if (options.disable) {
+              return false;
+            }
+
+            if (!Array.isArray(content) && content != null) {
+              throw new Error(
+                `Exported value was not extracted as an array: ${JSON.stringify(
+                  content
+                )}`
+              );
+            }
+
+            module[NS] = {
+              content,
+              options: opt || {},
+            };
+
+            return options.allChunks || module[`${NS}/extract`]; // eslint-disable-line no-path-concat
           };
-          return options.allChunks || module[`${NS}/extract`]; // eslint-disable-line no-path-concat
-        };
-      });
+        }
+      );
+
       const filename = this.filename;
       const id = this.id;
       let extractedChunks;
-      compilation.plugin('optimize-tree', (chunks, modules, callback) => {
-        extractedChunks = chunks.map(() => new Chunk());
-        chunks.forEach((chunk, i) => {
-          const extractedChunk = extractedChunks[i];
-          extractedChunk.index = i;
-          extractedChunk.originalChunk = chunk;
-          extractedChunk.name = chunk.name;
-          extractedChunk.setEntrypoints(chunk.getEntrypoints());
-          chunk.getChunks().forEach((c) => {
-            extractedChunk.addChunk(extractedChunks[chunks.indexOf(c)]);
-          });
-          chunk.getParents().forEach((c) => {
-            extractedChunk.addParent(extractedChunks[chunks.indexOf(c)]);
+      compilation.hooks.optimizeTree.tapAsync(
+        'optimize-tree',
+        (chunks, modules, callback) => {
+          extractedChunks = chunks.map(() => new Chunk());
+
+          chunks.forEach((chunk, i) => {
+            const extractedChunk = extractedChunks[i];
+            extractedChunk.index = i;
+            extractedChunk.originalChunk = chunk;
+            extractedChunk.name = chunk.name;
+
+            for (const chunkGroup of chunk.groupsIterable) {
+              if (chunkGroup instanceof Entrypoint) {
```

Comment:
Could you change this test to `chunkGroup.isInitial()`?

Reply:
I checked the implementation of ChunkGroup, isInitial() always returns false. There’s a deprecation warning that suggests using the instanceof Entrypoint check.